src/test/java/io/confluent/examples/streams/CustomJoinWithTableTriggeringStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,479 @@
+ * further down below as illustration of the implemented behavior.
+ *
+ * The approach in this example shares state stores between a stream-side and a table-side transformer.  This is safe
+ * because, if shared, Kafka Streams will place the transformers as well as the state stores into the same stream task,
+ * in which all access is exclusive and single-threaded.  The state store on the table side is the normal store of a
+ * KTable (thus avoiding data duplication due to store usage), whereas the state store on the stream side is manually
+ * added and attached to the processing topology.
+ *
+ * An alternative, more flexible approach is outlined further down below, in case you need additional control over the
+ * join behavior, e.g. by including stream-side vs. table-side timestamps in the decision-making logic.
+ *
+ * The default stream-table join behavior of Kafka Streams only triggers join output when data arrives at the stream
+ * side.  See https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Streams+Join+Semantics.
+ *
+ * Kafka Streams INNER stream-table join:
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | -
+ * 20   |                     ("alice", 1L) | -
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ * Kafka Streams LEFT stream-table join:
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | ("alice", (999.99, null)
+ * 20   |                     ("alice", 1L) | -
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ *
+ * The code in this example changes the above behavior so that an application will wait a configurable amount of time
+ * for data to arrive also at the table before it produces the join output for a given key (here: "alice").  The
+ * motivation is that, in this example, we'd prefer to receive fully populated join messages rather than join messages
+ * where the table-side information is missing (null).  Depending on your use case, you might prefer this changed
+ * behavior, and these changed semantics, over the default join semantics of Kafka Streams.
+ *
+ * Time | Stream              Table         | Join output
+ * -----+-----------------------------------+-------------------------
+ * 10   | ("alice", 999.99)                 | -
+ * 20   |                     ("alice", 1L) | ("alice", (999.99, 1L))
+ * 30   | ("alice", 555.55)                 | ("alice", (555.55, 1L))
+ *
+ * Note how, in the example above, the custom join produced a join output of `("alice", (999.99, 1L))`, even though the
+ * table-side record `("alice", 1L)` arrived AFTER the stream record `("alice", 999.99)`.
```

Comment:
`arrived AFTER` -- sound like processing time, not event-time to me. Thus, please indicate event-timestamps for the record (and make clear that `Time` column is processing-time.